lock-api/src/main/java/com/palantir/lock/client/TimelockRpcClient.java

```diff
@@ -0,0 +1,153 @@
+
+    @POST
+    @Path("fresh-timestamps")
+    TimestampRange getFreshTimestamps(@Safe @QueryParam("number") int numTimestampsRequested);
+
+    @POST
+    @Path("lock-immutable-timestamp")
+        // TODO (jkong): Can this be deprecated? Are there users outside of Atlas transactions?
+    LockImmutableTimestampResponse lockImmutableTimestamp(IdentifiedTimeLockRequest request);
+
+    /**
+     * @deprecated Please use {@link TimelockService#startIdentifiedAtlasDbTransaction(
+     *StartIdentifiedAtlasDbTransactionRequest)} instead; ignore the partition information if it is not useful for you.
+     */
+    @POST
+    @Path("start-atlasdb-transaction")
+    @Deprecated
+    StartAtlasDbTransactionResponse startAtlasDbTransaction(IdentifiedTimeLockRequest request);
+
+    @POST
+    @Path("start-identified-atlasdb-transaction")
+    StartIdentifiedAtlasDbTransactionResponse startIdentifiedAtlasDbTransaction(
+            StartIdentifiedAtlasDbTransactionRequest request);
+
+    @POST
+    @Path("immutable-timestamp")
+    long getImmutableTimestamp();
+
+    @POST
+    @Path("lock")
+    LockResponse lock(LockRequest request);
+
+    @POST
+    @Path("await-locks")
+    WaitForLocksResponse waitForLocks(WaitForLocksRequest request);
+
+    @POST
+    @Path("refresh-locks")
+    Set<LockToken> refreshLockLeases(Set<LockToken> tokens);
+
+    @POST
+    @Path("leasable-refresh-locks")
+    LeasableRefreshLockResponse leasableRefreshLockLeases(Set<LockToken> tokens);
+
+    @POST
+    @Path("leasable-lock")
+    LeasableLockResponse leasableLock(LockRequest request);
+
+    @POST
+    @Path("leasable-start-identified-atlasdb-transaction")
+    LeasableStartIdentifiedAtlasDbTransactionResponse leasableStartIdentifiedAtlasDbTransaction(
+            StartIdentifiedAtlasDbTransactionRequest request);
+
+    /**
+     * Releases locks associated with the set of {@link LockToken}s provided.
+     * The set of tokens returned are the tokens for which the associated locks were unlocked in this call.
+     * It is possible that a token that was provided is NOT in the returned set (e.g. if it expired).
+     * However, in this case it is guaranteed that that token is no longer valid.
+     *
+     * @param tokens Tokens for which associated locks should be unlocked.
+     * @return Tokens for which associated locks were unlocked
+     */
+    @POST
+    @Path("unlock")
+    Set<LockToken> unlock(Set<LockToken> tokens);
+
+    /**
+     * A version of {@link TimelockService#unlock(Set)} where one does not need to know whether the locks associated
+     * with the provided tokens were successfully unlocked or not.
+     *
+     * In some implementations, this may be more performant than a standard unlock.
+     *
+     * @param tokens Tokens for which associated locks should be unlocked.
+     */
+    default void tryUnlock(Set<LockToken> tokens) {
+        unlock(tokens);
+    }
+
+    @GET
+    @Path("leader-time")
+    IdentifiedTime getLeaderTime();
```

Comment:
it's worth a test that this changes every leadership election (on the same server), because this is part of the contract of timelock

Reply:
as in, if leadership goes A -> B -> A, A should have a different id the second time.